### PR TITLE
WIP: Prototype azure_core WASM support via reqwest

### PR DESCRIPTION
**Pull request by johnbatty**
In a recent hackathon I managed to get `azure_core` compiled to WASM and making requests from a browser.  This made use of `reqwest`'s WASM support but did require a couple of hacks to work around the fact that it doesn't yet support streaming responses (although there is an open PR which does appear to add this: https://github.com/seanmonstar/reqwest/pull/1576).

Not sure whether you want to take any part of this, but thought I'd make it available in case you are interested!

The main changes are:
- re-enable inclusion of `reqwest` and creation of a `reqwest` http client in the `http_client` module when the target is `wasm32`
- for `wasm32` target omit the `Send` trait when using `async_trait`
  - https://github.com/rustwasm/wasm-bindgen/issues/2409
- for `wasm32` target remove support for `Body::SeekableStream`
- when receiving a response read all the bytes and wrap it up in a "stream wrapper" (due to lack of a `bytes_stream(...)` function)
  - this is the hackiest part!  Any alternative suggestions welcomed!
- for `wasm32` target remove `Send + Sync` from the `PinnedStream` definition
- add a new feature `wasm-bindgen` that enables `time/wasm-bindgen`
  - if this `time` feature is not enabled then the retry policy code triggers a panic when querying the current time

### CODE REVIEW

**Comment by johnbatty** [comment]
Fyi, to get this working with my `azure-devops-rust-api` autorust-generated client I just had to replace `BoxFuture` with `LocalBoxFuture` to remove the `Send` trait:
https://github.com/microsoft/azure-devops-rust-api/pull/51/commits/d3e88b80770c4c69f5aba223e94cf55027355fe4#diff-54f4194d327855556893a578a400f781df3d8db6549897cc38d0e39da5ccf2bd

File: sdk/core/src/http_client/reqwest.rs [hunk 1]
```diff
@@ -24,6 +26,10 @@ impl HttpClient for ::reqwest::Client {
 
         let reqwest_request = match body {
             Body::Bytes(bytes) => req.body(bytes).build(),
+            // Hack... Temporarily remove for wasm32
+            // Need to implement properly when reqwest has support for streaming body.
+            // https://github.com/seanmonstar/reqwest/pull/1576
```
Comment (ctaggart) [inline sdk/core/src/http_client/reqwest.rs:31]: https://github.com/seanmonstar/reqwest/pull/1576 looks good to me, but has not been merged yet.

Comment (johnbatty) [reply sdk/core/src/http_client/reqwest.rs:31]: Looks like this function was merged to reqwest main last week via https://github.com/seanmonstar/reqwest/pull/1713.

Now just need to wait for it to appear in a release...

Comment (johnbatty) [reply sdk/core/src/http_client/reqwest.rs:31]: Ok, this is available in reqwest 0.11.14:
https://github.com/seanmonstar/reqwest/releases/tag/v0.11.14

I'll try updating the PR to use the new function...

**Comment by johnbatty** [comment]
Closing, as I've created a new PR with updated changes against the latest codebase: https://github.com/Azure/azure-sdk-for-rust/pull/1367

**Pull request closed without merging**